src/gui/dialogs/addon/addon_auth.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/addon/addon_auth.hpp"
+
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/password_box.hpp"
+#include "gui/widgets/window.hpp"
+
+namespace gui2::dialogs
+{
+
+REGISTER_DIALOG(addon_auth)
+
+addon_auth::addon_auth()
+{
+	password_ = register_text("password", true);
+}
+
+void addon_auth::pre_show(window& win)
+{
+	win.add_to_tab_order(find_widget<text_box>(&win, "password", false, true));
+}
+
+void addon_auth::post_show(window& win)
+{
+   password_text_ = find_widget<password_box>(&win, "password", false).get_real_value();
```

Comment:
Is it ok to keep the raw password value?

Reply:
Is there a reason it wouldn't be?